### ISSUE TITLE
This repository needs an integration test CI

### ISSUE THREAD

**Issue by galmasi** [keylime/attestation-operator#9]
Not sure where we can get our hands on a functional kube deployment with GHA. I will investigate.

**Comment by sarroutbi** [comment]
My proposal is to use Minikube. There are some deployments that can be done there. An example:
https://github.com/latchset/tang-operator/actions/runs/6264024941/job/17009764176

**Comment by mheese** [comment]
yeah, I've been thinking about this already as well. The difficulty is around what functionality we want to include with the integration tests. Particularly the operator relies on agents running that must have a TPM, a minikube setup unfortunately might not cut it.

**Comment by sarroutbi** [comment]
Sorry, I might not have explained correctly. The approach I propose would be something like:
1 - Install a host machine, and install minikube inside
2 - Start Verifier/Registrar (and any other component which is not the agent) in Minikube through attestation operator
3 - Run agents in the "host machine" (based host machine should support TPM)
4 - "Attack" Verifier/Registrar through Minikube (minikube ip will provide the host to connect to)
I will try to upload a Draft PR to demonstrate how to do so, just as Proof of Concept
